pkg/utils/compare_json.go

```diff
@@ -0,0 +1,150 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/olekukonko/tablewriter"
+)
+
+// JSONDiff represents the whole diff
+type JSONDiff struct {
+	Rows []JSONDiffRow
+}
+
+// JSONDiffRow represents a single non-existent subset item
+type JSONDiffRow struct {
+	Key      string
+	Expected interface{}
+	Got      interface{}
+}
+
+// PrintJSONDiff prints JSONDiff to STDERR
+func PrintJSONDiff(diff *JSONDiff) {
+	data := make([][]string, 0)
+
+	for _, r := range diff.Rows {
+		data = append(data, []string{r.Key, fmt.Sprintf("%s", r.Expected), fmt.Sprintf("%s", r.Got)})
+	}
+
+	table := tablewriter.NewWriter(os.Stderr)
+	table.SetHeader([]string{"Key", "Expected", "Got"})
+	for _, v := range data {
+		table.Append(v)
+	}
+	table.Render()
+}
+
+// JSONIsSubset checks if a is a subset json of b
+func JSONIsSubset(a, b []byte) (bool, *JSONDiff, error) {
+	return jsonIsSubsetR(a, b, nil, nil)
+}
+
+func jsonIsSubsetR(a, b []byte, diff *JSONDiff, prefix interface{}) (bool, *JSONDiff, error) {
```

Comment:
There are a lot of cases in here. Can we add some unit tests or is it covered in an integration test?

Reply:
@rhmdnd I just added integration test here: https://github.com/openshift/compliance-operator/pull/814/files#diff-0a9f36cc9b36d2e583d93013174fdc0bfedafad46ef1e2075effcbf8bbb87e8dR604

I am not sure if we could add an e2e test  for this one since it will be hard to mimic the delay for rendering KubeletConfig, but I think having this test should be enough. This fix will greatly improve the stability of our compliance operator.